### ISSUE TITLE
Issue with point 1.4.4 - text resizing up to 200%

### ISSUE THREAD

**Issue by makschernetskyi** [w3c/wcag#4730]
I would like to raise a concern regarding WCAG 2.1 and 2.2, point 1.4.4 (Resize Text, Level AA):

"1.4.4 Resize Text (Level AA) Except for captions and images of text:
text can be resized without assistive technology up to 200 percent without loss of content or functionality.”

The issue I encounter with this criterion is that it does not take into account the original text size of a platform.
For example, if headers are already large (for example 64 px), requiring them to remain fully legible at 128 px leads to impractical results, e.g. a single word occupying the entire screen, while still requiring the layout to remain functional.
Another example is small text. let's say for example some law firm decided to put small text, of size 1px at the end of some document. Well even if I increase the text size 200% I wouldn't be able to read such small text if I had some kind of vision impairment. But the website still would be WCAG 2.1 AA compliant, which also seems unreasonable.

Since WCAG serves as the foundation for legal standards such as the EAA and ADA, this limitation has significant implications.
I would therefore like to propose to refine this point and change it to something like this:

"1.4.4 Resize Text (Level AA) Except for captions and images of text:
text can be resized without assistive technology up to a legible physical size equivalent to 5 mm in character height, for at least the majority of screen resolutions within the range of 90 to 150 ppi, without loss of content or functionality.”

**Comment by airbr** [comment]
I was interested by your proposal and I note this has some history- https://github.com/w3c/wcag/issues/1671 . That GitHub issue has a lot of information. 

`e.g. a single word occupying the entire screen, while still requiring the layout to remain functional.` 

I would reasonably expect an application to remain functional even if this is happening, and even if the header wrapped some.

There exist a variety of CSS techniques to account for large headers? Responsive type?

**Comment by makschernetskyi** [comment]
> I would reasonably expect an application to remain functional even if this is happening, and even if the header wrapped some.
> 
> There exist a variety of CSS techniques to account for large headers? Responsive type?

To be honest, I would love if it would be so easy. however let's say I have an h1 in german - most popular language in EU - 64px large, twice that would be 128px which in my case means that most german words occupy twice the width of my screen. this means I would have to break words to make it functional, which means I have to detect this increase in textsize somehow and enable different type of breaking, since word breaking usually is a bad practice. such detection is a hard task for wordpress or wix pages etc. I mean it's possible to do so, everything is possible, but it is unreasonable to demand this.

**Comment by airbr** [comment]
You make good points about the difficulty, what I can say is a bunch of people are looking at this issue/have previously looked at it in the issue I mentioned https://github.com/w3c/wcag/issues/1671 . Internationalization is an interesting one because that presents layout issues already for most devs and requires effort to get right.

There are a number of blogs and articles - quite recent - talking about browsers approach to zooming text, example https://www.oddbird.net/2025/07/22/size-preferences/ . It seems there is still a lot of development in this general area that is happening and its not a super well solved problem. 

I am not a maintainer or long-time contributor so I dont know how much more to add, I hope some more engage with your question/ Ill add anything I think of.

**Comment by detlevhfischer** [comment]
This issue is well known - both the lack of a definition of a minimum size and the difficulties when applying the SC to text that is already large (as in headings).

Unless there is a normative change in WCAG 2.X that is not backwards compatible (which I believe is not currently planned), the issue will need to wait to be finally addressed in WCAG 3. 

MATF has a note which acknowledges one aspect of enlarging text that is already large: the non-linear scaling methods in iOS and Android. See https://github.com/w3c/matf/issues/3

WCAG2ICT has a similar note in 1.4.4 for non-web software: (Note 3 (Added) (for non-web software)) https://wcag2ict.netlify.app/#applying-sc-1-4-4-resize-text-to-non-web-software

**Comment by patrickhlauke** [comment]
part of the problem is, of course, that WCAG does not define any minimum sizes, or baseline sizes, or anything really. which is probaby good, as i fear otherwise we'd be stuck with some hardcoded values that may have been ok for 1999 but would now be quite outdated...

also, it obviously starts becoming very subjective when authors say their text is "already big enough" and that it doesn't need to let the user scale it any further.

**Comment by patrickhlauke** [comment]
>  legible physical size equivalent to 5 mm in character height

this sort of statement is mostly meaningless as authors have no actual control, nor knowledge, of how big things get rendered on the user's device (due to variations in screen sizes, pixel densities, OS settings, etc ... they CAN'T know this). it MAY work on a closed system where every device is the same (such as, say, specific terminals or point of sale touchscreens), but for the wild web, it's impossible (see also discussions from years ago about minimum target size, and how I pushed HARD on not enshrining a similar measure of "as rendered on screen" there, as it's impossible to control)

**Comment by GreggVan** [comment]
> part of the problem is, of course, that WCAG does not define any minimum sizes, or baseline sizes, or anything really. which is probaby good, as i fear otherwise we'd be stuck with some hardcoded values that may have been ok for 1999 but would now be quite outdated...
> 
> also, it obviously starts becoming very subjective when authors say their text is "already big enough" and that it doesn't need to let the user scale it any further.

Agree --  ALSO as mentioned above-- if you have no idea what platform is being used you have no idea what "size" it will be

**Comment by innoticFR** [comment]
Technique G178 alone defeats this guideline. If providing a control to change the size up to 200 percent is sufficient, and assuming this control is enabled by default, then merely providing a way to switch back to 100 percent would be sufficient.

**Comment by patrickhlauke** [comment]
@innoticFR I'm assuming "G78" is the wrong reference, as that's about audio description.

EDIT: Ah, found it ... you meant G178 https://www.w3.org/WAI/WCAG22/Techniques/general/G178

more generally, what do you mean by "defeats this guideline". do you mean "satisfies the requirements of this guideline", or...?

**Comment by innoticFR** [comment]
Indeed, it was G178. My mistake!

In this thread, the question was raised: what if the text is already enlarged?

Why couldn't one simply assume that the content currently displayed is already at 200% zoom by default? They would only need to implement a switch control (G178) and label the current state as '200%.'

This would completely defeat the purpose of the guideline while technically satisfying its requirements.

**Comment by patrickhlauke** [comment]
there are many example of malicious compliance. while they'd get away with claiming "it nominally passes WCAG", i doubt that if they get sued for having an inaccessible site that a judge/jury/panel would find in their favour...

**Comment by innoticFR** [comment]
In the example given by @makschernetskyi, the text is already large, meaning it would just be a way to comply to this guideline without doing anything. "Malicious" compliance, maybe, but legitimate.

The core issue arises when we translate WCAG guidelines into technical recommendations, as there is a critical element missing in Success Criterion 1.4.4.

It would be significantly easier to explain this guideline if the success criterion were worded as follows:

> Except for captions and images of text, text can be proportionally resized without assistive technology so that the smallest text size reaches 200 percent of the system default font size without loss of content or functionality. This requirement does not apply if the smallest text is already larger than this limit.

By clearly defining the system default font size as the baseline, this change would provide a fair and actionable metric: it would establish the truly expected minimum accessibility values while simultaneously avoiding the unnecessary requirement to double the size of already substantial text.

**Comment by patrickhlauke** [comment]
> It would be significantly easier to explain this guideline if ...

Yes, but we're not going to make a normative change of that magnitude anytime soon. So here and now, the requirement is that you can resize to 200% of the original default size, I'm afraid. Suggest bringing this up for WCAG3 consideration...

additionally, as a side note, "system default font size" is also variable, as users can change this in most cases in their OS settings.

**Comment by innoticFR** [comment]
Thanks, I understand that this would be a major change.

I am currently working on transcribing WCAG recommendations, and I wanted to ask: Do you personally (or anyone reading this) think that using the "system default font size" as a baseline could match the underlying intent of this guideline?

Additionally, could I suggest adding a simple note to WCAG 2.2? For example:

"Although the guideline does not specify a maximum font size value, the requested zoom factor is only meaningful when the initial font size is not already very large text."

I understand that overly small text is unusable regardless of the guideline's requirements

**Comment by detlevhfischer** [comment]
Two perhaps minor points: 

> "Although the guideline does not specify a maximum font size value, the requested zoom factor is only meaningful when the initial font size is not already very large text."

An important aspect of zoom is not just absolute size but the preservation of the intended hierarchy at different zoom levels. Authors may preserve such hierarchy in ways other than size (for example, using CSS to bolden or underline headings at a higher zoom rates)

> I understand that overly small text is unusable regardless of the guideline's requirements

In current browsers, zoom users can zoom to 500% on very small text so I would not consider it generally "unusable".

**Comment by patrickhlauke** [comment]
> Additionally, could I suggest adding a simple note to WCAG 2.2? For example:
>
> "Although the guideline does not specify a maximum font size value, the requested zoom factor is only meaningful when the initial font size is not already very large text."

adding that note would suggest that the SC can be ignored for some unspecified/subjective "very large" value. so I'm afraid that's not going to fly...

**Comment by GreggVan** [comment]
@patrickhlauke  
at first I thought that was a good idea (thinking of kiosks etc.).   But then I  thought of phones and I'm not sure what "very large" looks like on a standard sized smartphone.

**Comment by chaals** [comment]
@patrickhlauke said above:
> part of the problem is, of course, that WCAG does not define any minimum sizes, or baseline sizes, or anything really. which is probaby good, as i fear otherwise we'd be stuck with some hardcoded values that may have been ok for 1999 but would now be quite outdated...
> 
> also, it obviously starts becoming very subjective when authors say their text is "already big enough" and that it doesn't need to let the user scale it any further.

(See also the [following comment](https://github.com/w3c/wcag/issues/4730#issuecomment-3636199836), particularly
> this sort of statement is mostly meaningless as authors have no actual control, nor knowledge, of how big things get rendered on the user's device (due to variations in screen sizes, pixel densities, OS settings, etc ... they CAN'T know this).

I think we might have some space to manage a rational approach. Devices enable default font-size adjustment (that's why I haven't thrown my phone into the recycling bin). CSS enables measures that are "absolute" - `cm` or `pt` for example. Once upon a time I pushed hard not to allow those as values, but I think I wasn't getting it quite right.

The clear goal is to enable rendering at a pretty large font size - realistically, you can laugh but if "Schadenfreude" needs to be the only word on the screen to be large enough to be legible, then that's what is needed. What it means for the layout to "work" in such a scenario can be somewhat different to what I see in default layouts.

Sometimes, a linear layout loses important information (hello, tables), and that has to be re-established.

@GreggVan noted:
>  I'm not sure what "very large" looks like on a standard sized smartphone.

It looks like reading one word at a time. Makes pictures a kind of placeholder - you can see there is a picture, but if it's meaningful you probably want to see the alt text as soon as you scroll.

A phone with a screen less than 2" (5cm) diagonal size running Opera mini - I bought one of those 3 weeks ago - is going to have a pretty poor experience if the text size is set at 1.5cm (about 45pt). But then, text you cannot read is also a bad experience. I've watched people in many different circumstances and for many different reasons working through such experiences (and done it myself).

It might not be completely ridiculous to suggest that we set a minimum font-size at which things need to "work", knowing that it's a fudge (because different fonts look different at the same size, there are limits on what authors can control about the overall screen size, etc) but maybe one that we can get right enough to be useful in allowing people to read.

What we do about images is a whole question apart - insisting that it's OK to have an image at any size and have it include whatever people want without regard to whether people can make use of that is IMHO short-sighted. But teasing out the details is hard work that I think has never yet really been done properly.

**Comment by patrickhlauke** [comment]
> I think we might have some space to manage a rational approach. Devices enable default font-size adjustment (that's why I haven't thrown my phone into the recycling bin). CSS enables measures that are "absolute" - `cm` or `pt` for example.

noting that, per spec, on digital displays those absolute values are actually anchored on the CSS pixel, and generally don't bear any direct relation to the size something renders "as measured on the screen with a ruler". and even if you define things in `cm` etc, then go to print, the end result won't match it once you measure the printed output (I did an experiment on that particular aspect ages ago ... somewhere in the wiki for background information when we worked on target size, just to prove that point)

and yes, there have been various attempts/discussions in CSS space about "really" making some of these measurements match actual "as measured on screen" reality. and there's been discussions on "can we not just have an extra value that authors can access/use that gives them the zoom factor?" ... but all those butt up against the same problems

**Comment by alastc** [comment]
Noting that in WCAG3 the current equivalent requirement is:
> Text can be increased in size to at least 200% of the platform’s default body-text size.

That obviously has an assumption that there is a "platform default body-text size", which needs to be battle-tested, but I think it should work.

**Comment by patrickhlauke** [comment]
@alastc how does that then work for non-body text that's already sized up? e.g. if i have my `<h1>`s already set to `font-size: 200%` are they then exempt from needing to be resizable any further? what about preserving visual size hierarchy?

**Comment by alastc** [comment]
> [@alastc](https://github.com/alastc) how does that then work for non-body text that's already sized up? e.g. if i have my `<h1>`s already set to `font-size: 200%` are they then exempt from needing to be resizable any further? what about preserving visual size hierarchy?

Your H1 would meet the text-sizing requirement, but not the "Font-size not relied on" requirement: "When font size conveys visual meaning (such as headings), the text maintains its meaning and purpose when text is resized."

Just to note we got another issue about this in email:

> I'm extreemly glad to hear that the topic is already a concern.
> It truly breaks my UX heart to see rules like these causing so much damange.
> 
> But to answer your question.
> I would say that the heading size differences is less important than heading sizes breaking layout of the site entirely, and  im also surethat this can be solved with some math and css clamp magi.
> 
> So the value of heading size diffrences is significantly lower in terms of UX. Where having a heading with 320px would be a very bad experince for everyone and it is a gurantee it will disable all user to read headlines properly.
> 
> Anyway. I just had to vent my daily struggles of the manatory WCAG compliance standards that we HAVE to follow. 
> 
> Im just gonna leave this image here, to speak for itself. - Thanks for listening 🙂

<img width="797" height="1292" alt="screenshot showing a website with small ish text at the top, and one word split over three lines." src="https://github.com/user-attachments/assets/bd9447e2-e289-44dd-a7b2-e0708c698312" />
